packages/authorization-process/src/model/domain/errors.ts

```diff
@@ -371,6 +372,14 @@ export function eserviceNotDelegableForClientAccess(
   });
 }
 
+export function clientKindNotAllowed(clientId: ClientId): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Purpose operation not allowed for client ${clientId}`,
+    code: "clientKindNotAllowed",
+    title: "Purpose operation not allowed for client",
```

Comment:
`detail` and `title` refer to a "purpose operation" which is not indicated in the error name.
Also, the error is used in `assertClientIsConsumer` which is not strictly related to the purpose.

I suggest to make the detail and title generic on the client kind

```suggestion
    detail: `Client ${clientId} kind is not allowed for the requested operation`,
    code: "clientKindNotAllowed",
    title: "Operation not allowed on client kind",
```